quickwit/quickwit-search/src/collector.rs

```diff
@@ -303,20 +390,76 @@ pub struct QuickwitSegmentCollector {
     segment_ord: u32,
     timestamp_filter_opt: Option<TimestampFilter>,
     aggregation: Option<AggregationSegmentCollectors>,
-    search_after: Option<PartialHit>,
+    search_after: Option<SearchAfterSegment>,
     // Precomputed order for search_after for split_id and segment_ord
     precomp_search_after_order: Ordering,
 }
 
+/// Search After, but the sort values are converted to the u64 fast field representation.
+struct SearchAfterSegment {
+    sort_value: Option<u64>,
+    sort_value2: Option<u64>,
+    compare_on_equal: bool,
+    doc_id: DocId,
+}
+impl SearchAfterSegment {
+    fn new(
+        search_after: Option<PartialHit>,
+        score_extractor: &SortingFieldExtractorPair,
+    ) -> Option<Self> {
+        let Some(search_after) = search_after else {
+            return None;
+        };
+
+        let mut sort_value = None;
```

Comment:
Can `sort_value` ever be `None`?

Reply:
It will be `None` if the value on the passed value on the protobuf struct is `None`. 
Should we ignore that case? The comparison is on `Option<u64>`  anyway, since the fastfield may not have a value (except we add a specialized version for full columns)

Reply:
Ah yes if there are no search_after_sort_value.